translator/translate/otel/processor/metricsdecorator/translator.go

```diff
@@ -5,41 +5,96 @@ package metricsdecorator
 import (
 	"errors"
 	"fmt"
+	"strconv"
+	"strings"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/processor"
 
 	"github.com/aws/amazon-cloudwatch-agent/internal/metric"
+	"github.com/aws/amazon-cloudwatch-agent/internal/util/collections"
 	translatorcontext "github.com/aws/amazon-cloudwatch-agent/translator/context"
 	metricsconfig "github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/config"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/util"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
 )
 
-var metricsKey = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey)
-
 // ContextStatement follows the yaml structure defined by otel's transform processor:
 // https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/transformprocessor/internal/common/config.go#L45-L48)
 type ContextStatement struct {
 	Context    string   `mapstructure:"context"`
 	Statements []string `mapstructure:"statements"`
 }
 
+type Translator interface {
+	common.Translator[component.Config]
+	// IsSet determines whether the config has the fields needed for the translator.
+	IsSet(conf *confmap.Conf) bool
+}
+
 type translator struct {
-	name    string
-	factory processor.Factory
+	name          string
+	factory       processor.Factory
+	index         int
+	configKey     string
+	ignorePlugins collections.Set[string]
 }
 
-var _ common.Translator[component.Config] = (*translator)(nil)
+type transformFn = func(string) string
+
+type Option func(any)
+
+var (
+	defaultConfigKey = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey)
+)
+
+func WithName(name string) Option {
+	return func(a any) {
+		if t, ok := a.(*translator); ok {
+			t.name = name
+		}
+	}
+}
 
-func NewTranslator() common.Translator[component.Config] {
-	return NewTranslatorWithName("")
+func WithIndex(index int) Option {
+	return func(a any) {
+		if t, ok := a.(*translator); ok {
+			t.index = index
+		}
+	}
 }
 
-func NewTranslatorWithName(name string) common.Translator[component.Config] {
-	return &translator{name, transformprocessor.NewFactory()}
+func WithConfigKey(configKey string) Option {
+	return func(a any) {
+		if t, ok := a.(*translator); ok {
+			t.configKey = configKey
+		}
+	}
+}
+
+func WithIgnorePlugins(ignorePlugins ...string) Option {
```

Comment:
where is this being used? What plugins are we ignoring?

Reply:
@jefchien

Reply:
We're ignoring the JMX plugins in the host pipeline.